aiidalab/app.py

```diff
@@ -318,216 +298,164 @@ def _uninstall_app(self, _):
             cannot_modify = "you have local branches"
         else:
             # look for the local commited modifications all the available branches
-            initial_value = self.current_version
-            for key, value in self.available_versions.items():
-                self.version.selected.value = value  # switching to the branch in value
-                self._change_version(sleep_time=0)  # actually switching the branch
-                if self.found_local_commits():
-                    cannot_modify = "you have local commits ({} branch)".format(key)
-                    self.version.selected.value = initial_value  # switch back to the initial version
-                    self._change_version(sleep_time=0)  # actually switching the branch
-                    break
-
-        # And finally: uninstall or not?
-        if cannot_modify:
-            self.install_info.value = """<i class="fa fa-times" style="color:red;font-size:4em;" >
-            </i>Can not delete the repository: {}""".format(cannot_modify)
-            sleep(3)
+            with self._for_all_versions() as branches:
+                for branch in branches:
+                    if self._found_local_commits():
+                        raise RuntimeError("Can not delete the repository, there are local commits "
+                                           "on branch '{}'.".format(branch))
 
         # Perform uninstall process.
```

Comment:
We should actually convert all of the other branches in exception. I'll have another look at this and related functions, because there is a bunch of logic mixed and duplicated.